cmd/earthly/satellite_cmds.go

```diff
@@ -387,3 +399,132 @@ func (app *earthlyApp) actionSatelliteUnselect(cliCtx *cli.Context) error {
 
 	return nil
 }
+
+func (app *earthlyApp) actionSatelliteWake(cliCtx *cli.Context) error {
+	app.commandName = "satelliteWake"
+
+	if cliCtx.NArg() != 1 {
+		return errors.New("satellite name is required")
+	}
+
+	app.satelliteName = cliCtx.Args().Get(0)
+
+	cloudClient, err := cloud.NewClient(app.cloudHTTPAddr, app.cloudGRPCAddr, app.sshAuthSock, app.authToken, app.console.Warnf)
+	if err != nil {
+		return errors.Wrap(err, "failed to create cloud client")
+	}
+
+	orgID, err := app.getSatelliteOrgID(cliCtx.Context, cloudClient)
+	if err != nil {
+		return err
+	}
+
+	sat, err := cloudClient.GetSatellite(cliCtx.Context, app.satelliteName, orgID)
+	if err != nil {
+		return err
+	}
+
+	if sat.Status == cloud.SatelliteStatusOperational {
+		app.console.Printf("%s is already awake.", app.satelliteName)
+	}
+
+	out := cloudClient.WakeSatellite(cliCtx.Context, app.satelliteName, orgID)
+	err = showSatelliteLoading(app.console, app.satelliteName, out)
+	if err != nil {
+		return errors.Wrap(err, "failed waiting for satellite wake")
+	}
+
+	return nil
+}
+
+func showSatelliteLoading(console conslogging.ConsoleLogger, satName string, out chan cloud.SatelliteStatusUpdate) error {
+	loadingMsgs := getSatelliteLoadingMessages()
+	var (
+		loggedSleep      bool
+		loggedStop       bool
+		loggedStart      bool
+		shouldLogLoading bool
+	)
+	for o := range out {
+		if o.Err != nil {
+			return errors.Wrap(o.Err, "failed processing satellite status")
+		}
+		shouldLogLoading = true
+		switch o.State {
+		case cloud.SatelliteStatusSleep:
+			if !loggedSleep {
+				console.Printf("%s is waking up. Please wait...", satName)
+				loggedSleep = true
+				shouldLogLoading = false
+			}
+		case cloud.SatelliteStatusStopping:
+			if !loggedStop {
+				console.Printf("%s is falling asleep. Please wait...", satName)
```

Comment:
this message might be confusing to a user -- they want the satellite to wake up, yet the message is saying it's waiting for it to sleep. Maybe it should be pedantic to fully describe what is happening: `%s is falling asleep (once sleep event has completed) it will wake up. Please wait...`

Reply:
I updated this to say `%s is currently falling asleep. Waiting to send wake up signal...`  - I think it gets the right point across while still (somewhat) concise.